test/e2e/util.go

```diff
@@ -2789,3 +2789,51 @@ func getPodLogsInternal(c *client.Client, namespace, podName, containerName stri
 	}
 	return string(logs), err
 }
+
+// The following helper functions can block/unblock network from source host to destination host by manipulating iptable rules.
+// This function assumes it can ssh to the source host.
+// Suggested usage pattern:
+//
+// func foo() {
+//	...
+//	defer unblockNetwork(from, to)
+//	blockNetwork(from, to)
+//	...
+// }
+//
+func blockNetwork(from string, to string) {
+	Logf("block network traffic from %s to %s", from, to)
+	iptablesRule := fmt.Sprintf("OUTPUT --destination %s --jump REJECT", to)
+	dropCmd := fmt.Sprintf("sudo iptables --insert %s", iptablesRule)
+	if result, err := SSH(dropCmd, from, testContext.Provider); result.Code != 0 || err != nil {
```

Comment:
nit: probably 100% of the people who use this are going to poll till node NotReady right after, and then poll till NodeReady after recovery, so maybe encapsulate that too?